File: algorithm/binarySearch.py
def binary(arr, l, h, x):
    if h >= l:
        mid = l + (h - l) // 2
        
        # If found at mid, return mid
        if x == arr[mid]: 
            return mid
        
        # Search the right half
        elif x > arr[mid]:
            return binary(arr, mid + 1, h, x)  
        
        # Search the left half
        else:
            return binary(arr, l, mid - 1, x) 
    
    else:
        return -1

# Correct initialization with square brackets
arr = [2, 4, 7, 9, 12]
x = 9
    
result = binary(arr, 0, len(arr) - 1, x)
if result == -1:
    print("Element not found")
else:
    print("Element found at index " + str(result)) 

File: algorithm/binarySearch.php
<?php 

function binarySearch($arr, $target) {
    $low = 0; 
    $hei = sizeof($arr) - 1; 

    while ($low <= $hei) {
        $mid = intdiv($low + $hei, 2); 

        if ($arr[$mid] == $target) {
            return $arr[$mid];
        } else if ($arr[$mid] < $target) {
            // Search on the right side
            $low = $mid + 1;
        } else {
            // Search on the left side
            $hei = $mid - 1;
        }
    }

    return null; // Target not found
}

$arr = [2, 3, 4, 15, 26, 37];
$target = 4;
echo binarySearch($arr, $target);



?>